ui/webui/test/helpers/step_logger.py

```diff
@@ -0,0 +1,47 @@
+
+class BrowserSnapshot():
+    SNAPSHOT_NUMBER = 0
+
+    @classmethod
+    def new(cls, browser):
+        browser.snapshot(f'snapshot-{browser.label}', str(cls.SNAPSHOT_NUMBER))
+        cls.SNAPSHOT_NUMBER += 1
+
+
+def log_step(snapshots=False, snapshot_before=False, snapshot_after=False, docstring=False):
+    """ Decorator for logging steps durring testing.
+
+    Decorated function needs to be part of class with self.browser.
+
+    :param snapshots: Create snapshots before and after function call, defaults to False
+    :type snapshots: bool, optional
+    :param snapshot_before: Create snapshots before function call, defaults to False
+    :type snapshot_before: bool, optional
+    :param snapshot_after: Create snapshots after function call, defaults to False
+    :type snapshot_after: bool, optional
+    :param docstring: Print docstring of the function, defaults to False
+    :type docstring: bool, optional
+    """
+    def decorator(function):
+
+        def wrapper(*args, **kwargs):
+            nice_args = ', '.join(str(a) for a in args[1:])
+            if kwargs:
+                nice_args += ', ' + ', '.join(f'{k}: {v}' for k, v in kwargs.items())
+            if nice_args:
+                nice_args = ', with ' + nice_args
+
+            print(f'[TEST STEP] {function.__name__}{nice_args}')
+            if docstring:
+                print(f'[DOC] {function.__doc__}')
+            if snapshots or snapshot_before:
+                BrowserSnapshot.new(args[0].browser)
+            
+            function(*args, **kwargs)
```

Comment:
This function can have return value. Like for example [get_local_search](https://github.com/rhinstaller/anaconda/blob/master/ui/webui/test/helpers/language.py#L41) has. When you don't return it, python by default returns `None`. Which is why you see [here](https://github.com/rhinstaller/anaconda/pull/4575/files#diff-af148855d642cc07a588883e506315f3f74ff4f0ab87b832028ff650256ad626R69) `None`.

This simple fix makes the one failing test to pass:
```diff
--- ui/webui/test/helpers/step_logger.py
+++ ui/webui/test/helpers/step_logger.py
@@ -37,11 +37,13 @@ def log_step(snapshots=False, snapshot_before=False, snapshot_after=False, docst
             if snapshots or snapshot_before:
                 BrowserSnapshot.new(args[0].browser)
             
-            function(*args, **kwargs)
+            result = function(*args, **kwargs)
             
             if snapshots or snapshot_after:
                 BrowserSnapshot.new(args[0].browser)
 
+            return result
+
         return wrapper
     return decorator

```

Reply:
Good catch, thank you.